python/gui/auto_generated/qgsnewnamedialog.sip.in

```diff
@@ -80,6 +80,26 @@ Returns whether users are permitted to overwrite existing names.
 .. seealso:: :py:func:`setOverwriteEnabled`
 
 .. versionadded:: 2.12
+%End
+
+    void setAllowEmptyName( bool allowed );
+%Docstring
+Sets whether users are permitted to leave the widget empty.
+If ``True``, an autogenerated name is provided to the layout.
```

Comment:
QgsNewNameDialog is a generic class, so we shouldn't mention specifics like "is provided to the layout" here. Rather keep things generic like "the dialog will accept an empty name value".